lib/reek/source/source_code.rb

```diff
@@ -38,9 +41,9 @@ def initialize(code, description, parser = Parser::Ruby22)
       def self.from(source)
         case source
         when File     then new(source.read, source.path)
-        when IO       then new(source.readlines.join, 'STDIN')
+        when IO       then new(source.readlines.join, IO_IDENTIFIER)
         when Pathname then new(source.read, source.to_s)
-        when String   then new(source, 'string')
+        when String   then new(source, STRING_IDENTIFIER)
         end
```

Comment:
>> Is this really better?

I believe so.

>> We don’t use IO_IDENTIFIER nor STRING_IDENTIFIER anywhere else

True, but for me it's about intention revealing code in this case. For me the intention is "we use this literal for identifying xyz and this is will always be like that"
So for me it's about clarity and the detour, at least in this case, doesn't hurt that badly IMHO.

On a sidenote: 
I believe we should get rid of all magic strings and numbers in our source code, but this would be a different discussion.